samples/boards/sensortile_box_pro/ble-fw-upgrade/src/ble_fw.c

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+const uint8_t ble_fw_img[] = {
+	#include <dtm.bin.inc>
+};
```

Comment:
In order to avoid external dependencies, it may be possible to use a dummy include file with a predefined pattern instead of a real firmware image. Obviously, re-flashing BlueNRG-LP with a pattern will make the sensor tile box non- functional, but it will be useful to demonstrate that the sample works. This will remove the need of having an external blob binary file to run this example.

Reply:
Risk in that case is that someone runs the sample without the proper binary and then report that the sample doesn't have the expected effect. How to make sure we're able to quickly inform the user of missing dependency ?

Reply:
The other possibility is to add "skip: True" in sample.yaml, as you were suggesting. To be honest I saw that only a couple of samples are doing it, so I'm not sure it would be accepted.

Reply:
Problem here is that we're using samples concept (which is supposed to demonstrate code) as a utility to workaround a quite specific board issue.
Then, one thing about CI: CI primer goal primer goal is to test boards/SoC support to ensure that we detect issues quickly on drivers. Initially it didn't target samples as samples purpose was different. CI was added to samples just as n addition to existing coverage provided by test. And as a corollary it also help to verify that samples are always functional.

I'm stating all this to highlight that we shouldn't too much care about CI for this sample.
I think we should focus on the main purpose which is being utility that solves easily a problem for user. CI should not deviates us from this goal. If skipping it helps us in that goal, we should not hesitate.

Reply:
Added ```skip: true```

Reply:
> Risk in that case is that someone runs the sample without the proper binary and then report that the sample doesn't have the expected effect. How to make sure we're able to quickly inform the user of missing dependency ?

I think you can put a *big fat warning* in the console, when pattern is used, before asking to proceed with reflashing. In any case, I also agree that skipping CI can be an acceptable solution.